lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/config/LndConfig.scala

```diff
@@ -0,0 +1,268 @@
+  * different properties.
+  *
+  * Not all options are exposed from this class. We only
+  * expose those that are of relevance when making RPC
+  * requests.
+  */
+case class LndConfig(private[bitcoins] val lines: Seq[String], datadir: File)
+    extends Logging {
+
+  //create datadir and config if it DNE on disk
+  if (!datadir.exists()) {
+    logger.debug(
+      s"datadir=${datadir.getAbsolutePath} does not exist, creating now")
+    datadir.mkdirs()
+    LndConfig.writeConfigToFile(this, datadir)
+  }
+
+  private val confFile = datadir.toPath.resolve("lnd.conf")
+
+  //create lnd.conf file in datadir if it does not exist
+  if (!Files.exists(confFile)) {
+    logger.debug(
+      s"lnd.conf in datadir=${datadir.getAbsolutePath} does not exist, creating now")
+    LndConfig.writeConfigToFile(this, datadir)
+  }
+
+  /** Converts the config back to a string that can be written
+    * to file, and passed to `lnd`
+    */
+  lazy val toWriteableString: String = lines.mkString("\n")
+
+  /** Splits the provided lines into pairs of keys/values
+    * based on `=`, and then applies the provided
+    * `collect` function on those pairs
+    */
+  private def collectFrom(lines: Seq[String])(
+      collect: PartialFunction[(String, String), String]): Seq[String] = {
+
+    val splittedPairs = {
+      val splitLines = lines.map(
+        _.split("=")
+          .map(_.trim)
+          .toList)
+
+      splitLines.collect { case h :: t :: _ =>
+        h -> t
+      }
+    }
+
+    splittedPairs.collect(collect)
+  }
+
+  /** The blockchain network associated with this `lnd` config */
+  lazy val network: BitcoinNetwork = {
+    val isMainnet = getValue("bitcoin.mainnet").exists(_.toLowerCase == "true")
+    val isTestnet = getValue("bitcoin.testnet").exists(_.toLowerCase == "true")
+    val isSignet = getValue("bitcoin.simnet").exists(_.toLowerCase == "true")
+    val isRegtest = getValue("bitcoin.regtest").exists(_.toLowerCase == "true")
+
+    if (isMainnet && !(isRegtest || isSignet || isTestnet)) MainNet
+    else if (isTestnet && !(isRegtest || isSignet || isMainnet)) TestNet3
+    else if (isSignet && !(isRegtest || isTestnet || isMainnet)) SigNet
+    else if (isRegtest && !(isSignet || isTestnet || isMainnet)) RegTest
+    else
+      throw new IllegalArgumentException("No network config set in lnd.conf")
+  }
+
+  private[config] def getValue(key: String): Option[String] = {
+    val linesToSearchIn =
+      lines.takeWhile(l => !l.trim.startsWith("[") && !l.trim.startsWith(";"))
+    val collect = collectFrom(linesToSearchIn)(_)
+    collect { case (`key`, value) =>
+      value
+    }.headOption
+  }
+
+  /** Converts a string to an InetSocketAddress */
+  private def toInetSocketAddress(string: String): InetSocketAddress = {
+    val uri = new URI(string)
+    new InetSocketAddress(uri.getHost, uri.getPort)
+  }
+
+  lazy val bitcoindUser: String = getValue("bitcoind.rpcuser").getOrElse("foo")
```

Comment:
Probably should just use `.get` so that we throw in the case it isn't defined.

Let's fail fast rather than use a bad config that probably will fail else where.